src/primaite/agents/aegis/gllm.py

```diff
@@ -89,10 +96,62 @@ def get_cosine_embeddings_loss(self, openai_hidden_states, gllm_hidden_states):
 
         return loss
 
+    def get_crossentropy_loss(self, gt_probs, gt_tokens, gllm_probs, gllm_tokens):
+        criterion = torch.nn.CrossEntropyLoss(reduction="none")
+        padding_logit = torch.zeros([len(self.llm.tokenizer)])
+        padding_logit[self.llm.tokenizer.pad_token_id] = (
+            100  # Mock pad token of gt to 100% prob (it gets ignored anyway because of the mask)
+        )
+        padding_token = torch.tensor([self.llm.tokenizer.pad_token_id])  # Set padding token tensor
+
+        # Pad the shortest set of tensors if required
+        batch_size = gllm_probs.shape[0]
+        if gt_probs.shape[1] > gllm_probs.shape[1]:
+            # gt sequences are longer - pad gllm tokens
```

Comment:
Please re-write the comments now this is changed. As this is truncating the ground truth responses, no longer padding :)

Reply:
Also you can get rid of `n_pads` from this part directly below